modules/apps/foundation/user-associated-data/user-associated-data-test-util/src/main/java/com/liferay/user/associated/data/test/util/WhenHasStatusByUserIdField.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.user.associated.data.test.util;
+
+/**
+ * @author Noah Sherrill
+ */
+public interface WhenHasStatusByUserIdField {
+
+	public void addDataObjectWithStatusByUserId(
+			long userId, long statusByUserId)
+		throws Exception;
+
+}
```

Comment:
Would this class work as an inner class of the base test case?

Reply:
I'm not sure whether that could work. Some subclasses of the base test case will implement this interface, as BookmarksEntryUADEntityAggregatorTest does. It doesn't seem like a typical use case of an inner class. Also, the pattern used here was adopted from BaseTrashHandlerTestCase.

Reply:
Got it.  Thanks for explaining.